datalake/backend/processing/import_dataset/task.py

```diff
@@ -0,0 +1,142 @@
+import json
+import logging
+from datetime import datetime
+from http.client import responses as http_responses
+from os import environ
+from typing import Any, List, MutableMapping, Union
+from urllib.parse import urlparse
+from uuid import uuid4
+
+import boto3
+from jsonschema import ValidationError, validate  # type: ignore[import]
+from smart_open import open as smart_open  # type: ignore[import]
+
+from ..assets_model import ProcessingAssetsModel
+
+JsonList = List[Any]
+JsonObject = MutableMapping[str, Any]
+
+ssm_client = boto3.client("ssm")
+sts_client = boto3.client("sts")
+s3_client = boto3.client("s3")
+s3control_client = boto3.client("s3control")
+
+ENV = environ.get("DEPLOY_ENV", "test")
+STORAGE_BUCKET_PARAMETER = f"/{ENV}/storage-bucket-arn"
+S3_BATCH_COPY_ROLE_PARAMETER = f"/{ENV}/s3-batch-copy-role-arn"
+
+
+BODY_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "dataset_id": {"type": "string"},
+        "version_id": {"type": "string"},
+        "metadata_url": {"type": "string"},
+    },
+    "required": ["dataset_id", "metadata_url", "version_id"],
+}
+
+
+def lambda_handler(payload: JsonObject, _context: bytes) -> JsonObject:
+    """Main Lambda entry point."""
+
+    logger = set_up_logging()
+    logger.debug(json.dumps({"payload": payload}))
+
+    # validate input
+    try:
+        validate(payload, BODY_SCHEMA)
+    except ValidationError as error:
+        return error_response(400, error.message, logger)
+
+    dataset_id = payload["dataset_id"]
+    dataset_version_id = payload["version_id"]
+    metadata_url = payload["metadata_url"]
+
+    storage_bucket_arn = get_param(STORAGE_BUCKET_PARAMETER)
+    storage_bucket_name = storage_bucket_arn.rsplit(":", maxsplit=1)[-1]
+
+    staging_bucket_name = urlparse(metadata_url).netloc
+    manifest_key = f"manifests/{dataset_version_id}.csv"
+
+    with smart_open(f"s3://{storage_bucket_name}/{manifest_key}", "w") as s3_manifest:
+        for file in ProcessingAssetsModel.query(
```

Comment:
`file` is a very generic word. Also, `query` returns an iterable of items, so that might be a better name.